packages/graph/hash_graph/lib/graph/src/shared/identifier/time/axis.rs

```diff
@@ -0,0 +1,30 @@
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::identifier::time::{timestamp::Timestamp, version::VersionTimespan};
+
+/// Time axis for the decision time.
+///
+/// This is used as the generic argument to time-related structs and can be used as tag value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum DecisionTime {
+    #[default]
+    Decision,
+}
+
+pub type DecisionTimestamp = Timestamp<DecisionTime>;
+pub type DecisionTimeVersionTimespan = VersionTimespan<DecisionTime>;
```

Comment:
Is `Version` superfluous in this maybe? I _do_ like us having quite explicit self-explanatory code but `DecisionTimeVersionTimespan` is realllyyy wordy

Reply:
In https://github.com/hashintel/hash/pull/1738 I need a distinct `Timespan` struct, which contains different bounds. The version is guaranteed to always be `(Included, Excluded)` but for time projection defined in structural queries this does not apply. The different bounds are required as otherwise some timespans cannot be represented. With `(Included, Excluded)` a specific timestamp cannot be represented, without `(Included, Excluded)`, we cannot pass back `VersionTimespan` to the structural queries. Also `Unbounded` is required to select for a range until a certain timestamp or a range starting at a certain timestamp. Also, for getting all versions of an entity, `(Unbounded, Unbounded)` is required.
I elaborated a little bit more on the reasoning somewhere but I cannot find it currently 🙁

Reply:
I don't quite understand how that explains why we need the full length of the type name here, but we can revisit this as we tackle the stack of PRs.

Reply:
With `Timespan` added as well it's likely that we will also have a type alias for `DecisionTimespan` or something similar. However, It's also likely that we don't need any of these aliases later on. Yes, I will revisit this later.